src/libraries/Native/Unix/System.Native/pal_io.c

```diff
@@ -370,7 +370,7 @@ int32_t SystemNative_GetReadDirRBufferSize(void)
 {
 #if HAVE_READDIR_R
     // dirent should be under 2k in size
-    assert(sizeof(struct dirent) < 2048);
+    static_assert(sizeof(struct dirent) < 2048);
```

Comment:
Shoot. Sorry @stephentoub, my bad. I didn't realized this is a C file, [`static_assert`](https://en.cppreference.com/w/cpp/language/static_assert) is only for C++. This needs to be an `assert`.